pandas/io/xml.py

```diff
@@ -0,0 +1,1017 @@
+                    r = parse(f, parser=curr_parser)
+            elif is_xml:
+                r = XML(bytes(current_doc, encoding=self.encoding))
+            else:
+                r = parse(current_doc, parser=curr_parser)
+        except (LookupError, URLError, HTTPError, OSError, XMLSyntaxError) as e:
+            raise e
+
+        return r
+
+
+def _data_to_frame(data, **kwargs) -> DataFrame:
+    """
+    Convert parsed data to Data Frame.
+
+    This method will bind xml dictionary data of keys and values
+    into named columns of Data Frame using the built-in TextParser
+    class that build Data Frame and infers specific dtypes.
+    """
+
+    tags = [list(d.keys()) for d in data]
+    nodes = [list(d.values()) for d in data]
+
+    try:
+        with TextParser(nodes, names=tags[0], **kwargs) as tp:
+            return tp.read()
+    except ParserError:
+        raise ParserError(
+            "XML document may be too complex for import. "
+            "Try to flatten document and use distinct "
+            "element and attribute names."
+        )
+
+
+def _parse(
+    io,
+    xpath,
+    namespaces,
+    elems_only,
+    attrs_only,
+    names,
+    encoding,
+    parser,
+    stylesheet,
+    **kwargs,
+) -> DataFrame:
+    """
+    Call internal parsers.
+
+    This method will conditionally call internal parsers:
+    LxmlFrameParser and/or EtreeParser.
+
+    Raises
+    ------
+    ValueError
+        * If parser is not lxml or etree.e.
+
+    Notes
+    -----
+    This method will raise a warning instead of module not found or
+    import error if user does not have 1xml and then reverts to
+    fallback option with etree parser.
+    """
+
+    if parser == "lxml":
+        try:
+            p = _LxmlFrameParser(
+                io,
+                xpath,
+                namespaces,
+                elems_only,
+                attrs_only,
+                names,
+                encoding,
+                stylesheet,
+            )
+        except ImportError:
+            warn(
+                "You do not have lxml installed (default parser). "
+                "Instead, etree will be used.",
+                ImportWarning,
+            )
+
+            p = _EtreeFrameParser(
+                io,
+                xpath,
+                namespaces,
+                elems_only,
+                attrs_only,
+                names,
+                encoding,
+                stylesheet,
+            )
+
+    elif parser == "etree":
+        p = _EtreeFrameParser(
+            io,
+            xpath,
+            namespaces,
+            elems_only,
+            attrs_only,
+            names,
+            encoding,
+            stylesheet,
+        )
+    else:
+        raise ValueError("Values for parser can only be lxml or etree.")
+
+    data_dicts = p.parse_data()
+
+    return _data_to_frame(data=data_dicts, **kwargs)
+
+
+@deprecate_nonkeyword_arguments(version="2.0")
```

Comment:
you don't need to do this, just make everything keyword only (except for io) but rename that to path_or_buf